src/sentry_core.c

```diff
@@ -175,6 +175,19 @@ sentry_capture_event(sentry_value_t event)
     }
     if (opts->transport && !sentry_value_is_null(event)) {
         sentry_envelope_t *envelope = sentry__envelope_new();
+
+        SENTRY_TRACE("adding attachments to envelope");
+        for (sentry_attachment_t *attachment = opts->attachments; attachment;
+             attachment = attachment->next) {
+            sentry_envelope_item_t *item = sentry__envelope_add_from_path(
+                envelope, attachment->path, "attachment");
+            if (!item) {
+                sentry_envelope_free(envelope);
```

Comment:
If you free the envelope here, then you should probably bail the function. However, it seems that you might just want to skip this item since this can totally happen if the file pointed to doesn't exist. This is to be expected.